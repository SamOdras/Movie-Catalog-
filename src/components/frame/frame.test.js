import Frame from './frame.component';
import {
  render as testRender,
  fireEvent,
  screen
} from "@testing-library/react";

describe('Render without crashing', () => {
  const props = {
    isLoading: false,
    searchValue: "",
    onChangeSearch: jest.fn(),
    children: <></>
  };
  const render = () => testRender(<Frame {...props} />);

  test('Render Frame', () => {
    expect(render()).toMatchSnapshot();
  })

  test("Call on change search, when text change", () => {
    render();
    const input = screen.getByTitle("searchMovie");
    fireEvent.change(input, { target: { value: "Hello There" } });
    expect(props.onChangeSearch).toHaveBeenCalledTimes(1);
  });

  test("Click title", () => {
    render();
    const title = screen.getByTitle("brand");
    fireEvent.click(title)
    expect(window.location.href).toBe("http://localhost/");
  })
  test("Expecting boolean", () => {
    const { container } = render();
    expect(container.querySelector('[title="loading-indicator"]')).toBeNull();
  })
})
