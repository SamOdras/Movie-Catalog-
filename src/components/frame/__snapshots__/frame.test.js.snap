// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render without crashing Render Frame 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="navbar navbar-expand navbar-light bg-light fixed-top"
          style="box-shadow: 1px 4px 4px -4px; height: 60px;"
        >
          <div
            class="container"
          >
            <a
              class="navbar-brand"
              style="cursor: pointer;"
              title="brand"
            >
              Movie Catalog
            </a>
            <button
              aria-label="Toggle navigation"
              class="navbar-toggler"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="collapse navbar-collapse"
            >
              <ul
                class="me-auto navbar-nav"
              >
                <input
                  class="form-control"
                  placeholder="Cari Film"
                  style="margin-left: 20px; width: 500px;"
                  title="searchMovie"
                  type="search"
                  value=""
                />
              </ul>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="navbar navbar-expand navbar-light bg-light fixed-top"
        style="box-shadow: 1px 4px 4px -4px; height: 60px;"
      >
        <div
          class="container"
        >
          <a
            class="navbar-brand"
            style="cursor: pointer;"
            title="brand"
          >
            Movie Catalog
          </a>
          <button
            aria-label="Toggle navigation"
            class="navbar-toggler"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
          >
            <ul
              class="me-auto navbar-nav"
            >
              <input
                class="form-control"
                placeholder="Cari Film"
                style="margin-left: 20px; width: 500px;"
                title="searchMovie"
                type="search"
                value=""
              />
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
